packages/storage/src/storage.ts

```diff
@@ -473,7 +475,11 @@ export function createStorageSignal<T, O = {}>(
       }
       return value;
     }, null);
-  const [accessor, setter] = createSignal<T | null>(read() ?? (initialValue as T), props as any);
+  const [accessor, setter] = createHydrateSignal<T | null>(
+    initialValue as T,
+    () => read() ?? (initialValue as T),
```

Comment:
This way, initialValue will also be returned if the signal's value is `null`. This is not the intended behavior.

Reply:
I disagree, it was implemented this way.

When storage is empty (or SSR): `read()` return `null` therefore it should set the storage.key to the `initialValue` (or return it during SSR).

Then, if you call the `setter` to set the value to `null` it will remove the storage.key

Reply:
Thanks for the clarification. Sorry, I had only my mobile to check on and there must have been an error with StackBlitz.